pkg/storage/storage.go

```diff
@@ -17,38 +17,46 @@ limitations under the License.
 package storage
 
 import (
+	"fmt"
+
 	"github.com/openebs/openebsctl/pkg/client"
 	"github.com/openebs/openebsctl/pkg/util"
-	"github.com/pkg/errors"
 )
 
 // Get manages various implementations of Storage listing
 func Get(pools []string, openebsNS string, casType string) error {
-	// TODO: Change the below implementation once more castype pool listing is in place
-	if casType != util.CstorCasType && casType != "" {
-		return errors.Errorf("storage listing feature not available for %s", casType)
-	}
 	// 1. Create the clientset
 	k, _ := client.NewK8sClient("")
-	// 2. Get the namespaces
-	nsMap, _ := k.GetOpenEBSNamespaceMap()
-	if openebsNS == "" {
-		if val, ok := nsMap[casType]; ok {
-			k.Ns = val
-		}
+	// 2. If casType is specified, call the specific function & exit
+	if f, ok := CasListMap()[casType]; ok {
+		// if a cas-type is found, run it and return the error
+		return f(k, pools)
 	}
-	// TODO: Change this line, currently this overwriting empty flag values as cstor
-	if list, ok := CasListMap()[util.CstorCasType]; ok {
-		err := list(k, pools)
-		if err != nil {
-			return err
+	// 3. Call all functions & exit
+	var separator bool
+	for _, f := range CasList() {
+		if separator {
+			fmt.Println()
+		}
+		err := f(k, pools)
+		if err == nil {
+			// A visual separator for different cas-type pools/storage entities
+			separator = true
+		} else {
+			separator = false
 		}
 	}
 	return nil
 }
 
+// CasList has a list of method implementations for different cas-types
+func CasList() []func(*client.K8sClient, []string) error {
+	return []func(*client.K8sClient, []string) error{
+		GetCstorPools, GetVolumeGroups, GetZFSNodes}
```

Comment:
Just a nit, Can we call the method GetZFSPools as we are calling GetVolumeGroups for lvm. Its referring to the pool/storage and not the CR.